File: backend/src/main/java/br/com/surb/mysales/services/client/ClientFindAllService.java
package br.com.surb.mysales.services.client;

import br.com.surb.mysales.dto.client.ClientDTO;
import br.com.surb.mysales.entities.Client;
import br.com.surb.mysales.mapper.ClientMapper;
import br.com.surb.mysales.repositories.ClientRepository;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class ClientFindAllService {
  private final ClientRepository clientRepository;

  public ClientFindAllService(ClientRepository clientRepository) {
    this.clientRepository = clientRepository;
  }

  @Transactional(readOnly = true)
  public Page<ClientDTO> execute(Pageable pageable) {
    Page<Client> response = clientRepository.findAll(pageable);
    return response.map(client -> ClientMapper.toDTO(client));
  }
}

File: frontend/src/services/client/ClientFindAll.service.ts
import { api } from '@/http/api';
import { ClientDTO } from '@/http/response/ClientDTO';
import { PageResponse } from '@/http/response/PageResponse';
import { AxiosResponse } from 'axios';

export const ClientFindAllService = async (): Promise<
  PageResponse<ClientDTO>
> => {
  const response: AxiosResponse<PageResponse<ClientDTO>> = await api.get(
    `/v1/clients`
  );
  return response.data;
};
